sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -105,10 +105,30 @@ abstract class JdbcDialect extends Serializable with Logging {
    */
   def getJDBCType(dt: DataType): Option[JdbcType] = None
 
+  /**
+   * Converts an instance of `java.sql.Timestamp` to the number of microseconds since
+   * 1970-01-01T00:00:00.000000Z. It extracts date-time fields from the input, builds
+   * a local timestamp in Proleptic Gregorian calendar from the fields, and binds
+   * the timestamp to the system time zone. The resulted instant is converted to
+   * microseconds since the epoch.
+   * JDBC dialects can override this function to provide implementations that suit their
+   * JDBC drivers (e.g. if there are special "infinity" values that would overflow)
+   *
+   * @param t represents a specific instant in time based on
+   *          the hybrid calendar which combines Julian and
+   *          Gregorian calendars.
+   * @return The number of micros since epoch from `java.sql.Timestamp`.
+   * @throws IllegalArgumentException if t is null
+   */
+  def convertJavaTimestampToTimestamp(t: Timestamp): Long = {
+    require(t != null, "timestamp must be non-null")
```

Comment:
Is this an user-facing error? If so, please, add an error class to `error-classes.json`. See for instance:
https://github.com/apache/spark/blob/7bc28d54f83261b16eaa11201a7987d8d2c8dd1e/sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala#L478

Reply:
It is not an user facing error but is a precondition: the timestamp passed into this function will never be null. I removed the null check because a similar function `convertJavaTimestampToTimestampNTZ` in JdbcDialects.scala does not have the check either.